src/module/actor/util.ts

```diff
@@ -168,19 +169,34 @@ export async function update_pilot(pilot: LancerActor, cc_pilot: mm.Pilot): Prom
 
         return mount
       })
-      
+
       mounts = integratedMounts.concat(equippableMounts)
 
       pd.mech_loadout.mounts = mounts
-      
+
     }
-    
+
+    // Setup token
+    let mech_image = am.CloudImage || get_mech_art(am.Frame.ID) || pilot.data.img;
+    // FIXME: Setting type as any to allow editing the token attributes
+    let ptoken = duplicate(pilot.data.token) as any;
+
+    pad.img = mech_image;
+    ptoken.name = cc_pilot.Callsign;
+    ptoken.img = mech_image;
```

Comment:
This is unconditionally overwriting the token image. This should only happen if the token is already using a default image.